src/standalone.js

```diff
@@ -21,7 +21,27 @@ try {
   window.RUM_BASE = window.RUM_BASE || scriptSrc;
   window.RUM_PARAMS = window.RUM_PARAMS || scriptParams;
 
-  sampleRUM();
+  let status;
+  if (window.performance && window.performance.getEntriesByType) {
+    for (const { name, responseStatus, serverTiming } of window.performance.getEntriesByType("navigation")) {
+      if (name === window.location.href) {
+        status = responseStatus;
+      }
+      if (!status && serverTiming) {
+        for (const {name, duration} of serverTiming) {
+          if (name === "status") {
+            status = duration;
+          }
+        }
+      }
+    }
+  }
+
+  if (status && status === 404) {
+    sampleRUM('404', { source: document.referrer });
+  } else {
+    sampleRUM();
+  }
```

Comment:
`serverTiming` is populated from Server-Timing response headers which are usually not set, so it's usually empty. When the header is set, we do see it there even in Safari.
But if we're fine not supporting Safari, which I think is fine for 404 tracking, because I don't expect a different set of 404 links on Safari specifically, then we don't need to look for `serverTiming`.
Also since we only expect one navigation entry, we can skip the `navigation.name === window.location.href` check, since it should always be for the current page.
```suggestion
const [navigation] = (window.performance && window.performance.getEntriesByType("navigation")) || [];
if (navigation && navigation.responseStatus === 404) {
    sampleRUM('404', { source: document.referrer });
} else {
    sampleRUM();
}
```